src/components/Store.react.js

```diff
@@ -186,11 +153,19 @@ export default class Store extends React.Component {
         }
         const old = this._backstore.getItem(id);
         // Only set the data if it's not the same data.
-        if (dataChanged(data, old)) {
-            this._backstore.setItem(id, data);
-            setProps({
-                modified_timestamp: this._backstore.getModified(id),
-            });
+        // If the new data is undefined, we got here by overwriting the entire
+        // component with a new copy that has no `data` specified - so pull back
+        // out the old value.
+        // Note: this still allows you to set data to null
+        if (!equals(data, old)) {
+            if (data === undefined) {
```

Comment:
I bet this would all be even simpler if we moved to a functional component with `useEffect`, but anyway this is the key point allowing you to re-supply the same component with no `data` and have the old value persist.

If you actually remove it completely and bring it back later, it already worked, via the `UNSAFE_componentWillMount` pathway. This is only if the component is already on the page and you provide it again - ie as part of a larger `children` prop in which some other things may have changed.